nautobot_ssot/integrations/aristacv/utils/cloudvision.py

```diff
@@ -253,17 +253,20 @@ def get_devices(client, import_active: bool):
         req = services.DeviceStreamRequest()
     responses = device_stub.GetAll(req)
     devices = []
-    for resp in responses:
-        device = {
-            "device_id": resp.value.key.device_id.value,
-            "hostname": resp.value.hostname.value,
-            "fqdn": resp.value.fqdn.value,
-            "sw_ver": resp.value.software_version.value,
-            "model": resp.value.model_name.value,
-            "status": "Active" if resp.value.streaming_status == 2 else "Offline",
-            "system_mac_address": resp.value.system_mac_address.value,
-        }
-        devices.append(device)
+    try:
+        for resp in responses:
+            device = {
+                "device_id": resp.value.key.device_id.value,
+                "hostname": resp.value.hostname.value,
+                "fqdn": resp.value.fqdn.value,
+                "sw_ver": resp.value.software_version.value,
+                "model": resp.value.model_name.value,
+                "status": "Active" if resp.value.streaming_status == 2 else "Offline",
+                "system_mac_address": resp.value.system_mac_address.value,
+            }
+            devices.append(device)
+    except grpc.RpcError as err:
```

Comment:
what is raising the `grpc.RpcError`? I thought it could be `device_stub.GetAll(req)` instead of the response value access...

Reply:
@chadell I thought the same thing but when I put the block around that it didn't actually catch it. It seems to be only when you're iterating through the messages, or I guess viewing the stream, that the error is thrown. In my particular test case it was an authentication error though I'm not sure what the other opened issue is having the error from.